src/StaticLS/IDE/Definition.hs

```diff
@@ -0,0 +1,115 @@
+module StaticLS.IDE.Definition where
+
+import GHC.Plugins
+import qualified HieDb
+import qualified Language.LSP.Types as LSP
+import StaticLS.HIE
+import StaticLS.HIE.File
+import StaticLS.Monad
+import System.FilePath ((</>))
+import Control.Monad.Trans.Maybe
+import qualified GHC.Data.FastString as GHC
+import qualified GHC.Iface.Ext.Types as GHC
+import Data.List (isSuffixOf)
+import System.Directory (doesFileExist)
+import Control.Monad (join, guard)
+import GHC.Utils.Monad (mapMaybeM)
+import Control.Monad.Trans.Class (lift)
+import Data.Maybe (maybeToList, fromMaybe)
+import Development.IDE.GHC.Error
+  ( srcSpanToRange
+  , srcSpanToFilename
+  )
+
+locationsAtPoint ::
+  HasStaticEnv m =>
+  LSP.TextDocumentIdentifier ->
+  LSP.Position ->
+  m [LSP.Location]
+locationsAtPoint tdi pos = do
+  mLocs <- runMaybeT $ do
+    hieFile <- MaybeT $ getHieFileFromTdi tdi
+    let identifiersAtPoint = join $
+              HieDb.pointCommand
+                hieFile
+                (lspPositionToHieDbCoords pos)
+                Nothing
+                hieAstNodeToIdentifiers
+    join <$> mapM (MaybeT . fmap Just . identifierToLocation) identifiersAtPoint
+  pure $ fromMaybe [] mLocs
+    where 
+      identifierToLocation :: HasStaticEnv m => GHC.Identifier -> m [LSP.Location]
+      identifierToLocation =
+        either
+          (fmap maybeToList . modToLocation)
+          nameToLocation
+
+      modToLocation :: HasStaticEnv m => ModuleName -> m (Maybe LSP.Location)
+      modToLocation modName =
+        let zeroPos = LSP.Position 0 0
+            zeroRange = LSP.Range zeroPos zeroPos in
+              runMaybeT $ do
+                srcFile <- MaybeT (modToSrcFile modName)
+                pure $ LSP.Location (LSP.filePathToUri srcFile) zeroRange
+
+---------------------------------------------------------------------
+  -- The following code is largely taken from ghcide with slight modifications
+---------------------------------------------------------------------
+
+-- | Given a 'Name' attempt to find the location where it is defined.
```

Comment:
Most of this is pretty directly taken from ghcide with very slight modifications to fit the `HasStaticEnv` monad + change up the filepath a bit

Reply:
Thanks, added to both this and the halfsp one!